index.js

```diff
@@ -929,6 +929,21 @@
       [Fn(a, b), Fn(c, d), p(a, c), p(b, d)],
       Z.bimap);
 
+  //# mapLeft :: Bifunctor f => (a -> b) -> f a c -> f b c
+  //.
+  //. Curried version of [`Z.mapLeft`][]. Maps the given function over the left
+  //. side of a Bifunctor.
+  //.
+  //. ```javascript
+  //. > S.mapLeft(S.toUpper, S.Left('foo'))
+  //. Left('FOO')
+  //.
+  //. > S.mapLeft(S.toUpper, S.Right(64))
+  //. Right(64)
+  //. ```
+  S.mapLeft =
+  def('mapLeft', {p: [Z.Bifunctor]}, [Fn(a, b), p(a, c), p(b, c)], Z.mapLeft);
```

Comment:
We're using the “wrong” letter for the binary type variable, because `f` is defined as a unary type variable. We *could* do this:

```javascript
S.mapLeft = (function() {
  var f = $.BinaryTypeVariable('f');
  return def('mapLeft', ...);
}());
```

This strikes me as quite ugly, so I'm inclined to stick with the status quo. If one day we decide to define each function in a separate file we could switch to the “correct” letters at that point. I'll continue to fight to keep the Sanctuary source code a single ES5 file, of course. :)

Reply:
I think it's fine. Give `p` a chance to shine.

Reply:
I also think `p` is fine. Assuming `s` and `p` stand for Semigroupoid and Profunctor, respectively, you could always define `bi` for Bifunctor.

Another option would be to get rid of `s` and just use `p` for Semigroupoid, Profunctor, and Bifunctor